flink-connector-hbase-2.2/src/test/java/org/apache/flink/connector/hbase2/HBaseConnectorITCase.java

```diff
@@ -451,6 +452,60 @@ public void testTableSinkWithTimestampMetadata() throws Exception {
         TestBaseUtils.compareResultAsText(results, expected);
     }
 
+    @Test
+    public void testTableSinkWithTTLMetadata() throws Exception {
+        StreamExecutionEnvironment execEnv = StreamExecutionEnvironment.getExecutionEnvironment();
+        StreamTableEnvironment tEnv = StreamTableEnvironment.create(execEnv, streamSettings);
+
+        tEnv.executeSql(
+                "CREATE TABLE hTableForSink ("
+                        + " rowkey INT PRIMARY KEY NOT ENFORCED,"
+                        + " family1 ROW<col1 INT>,"
+                        + " ttl BIGINT NOT NULL METADATA FROM 'ttl'"
+                        + ") WITH ("
+                        + " 'connector' = 'hbase-2.2',"
+                        + " 'table-name' = '"
+                        + TEST_TABLE_6
+                        + "',"
+                        + " 'zookeeper.quorum' = '"
+                        + getZookeeperQuorum()
+                        + "'"
+                        + ")");
+
+        String insert =
+                "INSERT INTO hTableForSink VALUES"
+                        + "(1, ROW(1), 4000),"
+                        + "(2, ROW(2), 3500),"
+                        + "(3, ROW(3), 5000)";
+        tEnv.executeSql(insert).await();
+
+        tEnv.executeSql(
+                "CREATE TABLE hTableForQuery ("
+                        + " rowkey INT PRIMARY KEY NOT ENFORCED,"
+                        + " family1 ROW<col1 INT>"
+                        + ") WITH ("
+                        + " 'connector' = 'hbase-2.2',"
+                        + " 'table-name' = '"
+                        + TEST_TABLE_6
+                        + "',"
+                        + " 'zookeeper.quorum' = '"
+                        + getZookeeperQuorum()
+                        + "'"
+                        + ")");
+        String query = "SELECT rowkey, family1.col1 FROM hTableForQuery";
+
+        TableResult firstResult = tEnv.executeSql(query);
+        List<Row> firstResults = CollectionUtil.iteratorToList(firstResult.collect());
+        String firstExpected = "+I[1, 1]\n+I[2, 2]\n+I[3, 3]\n";
+        TestBaseUtils.compareResultAsText(firstResults, firstExpected);
+
+        TimeUnit.SECONDS.sleep(6);
+
+        TableResult lastResult = tEnv.executeSql(query);
+        List<Row> lastResults = CollectionUtil.iteratorToList(lastResult.collect());
+        assertThat(lastResults).isEmpty();
```

Comment:
(Same for the hbase-1.4 test)

I think it would worth to have another read at a time where there is still some records in the table, e.g.:
```java
TableResult tRes = tEnv.executeSql(query);
List<Row> actualRows = CollectionUtil.iteratorToList(tRes.collect());
String expectedRows = "+I[1, 1]\n+I[2, 2]\n+I[3, 3]\n";
TestBaseUtils.compareResultAsText(actualRows, expectedRows);

Thread.sleep(4200);

tRes = tEnv.executeSql(query);
actualRows = CollectionUtil.iteratorToList(tRes.collect());
expectedRows = "+I[3, 3]";
TestBaseUtils.compareResultAsText(actualRows, expectedRows);

Thread.sleep(2000);

tRes = tEnv.executeSql(query);
actualRows = CollectionUtil.iteratorToList(tRes.collect());
assertThat(actualRows).isEmpty();
```